### PR TITLE
Update JavaScript : Arrow functions

### LINKED ISSUE #1378: [Edit] JavaScript: Arrow Functions
### Type of Edit (select all that apply)

Fix grammar, spelling, formatting

### Description (optional)

Could we make the following edits to the `arrow-functions.md` file, inside the `content/javascript/concepts/arrow-functions/` directory?

- On line # 17, either change to "\*\*Arrow functions\*\* are expressions first introduced...." _or_ "\*\*Arrow function\*\* expressions were first introduced..."
- The list items on lines # 21 & 22 should be moved somewhere in the "Syntax" section below (preferably below line # 34.
- Lines # 36-41: the block quote **note** should be changed into its own section with a lv. 3 header titled "Limitations".
- Line # 71: The code block signature needs to be changed to \```codebyte/javascript.

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by faiz887**
### Description

Closes #1378 

### Type of Changes :

- Fixed the grammar mistake on line 17
- Formatted the list items on line 21 & 22 and put it under the section of syntax
- Made a new section called Limitations instead of it being under note
- Changed the code block signature

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/javascript/concepts/arrow-functions/arrow-functions.md [hunk 1]
```diff
@@ -33,16 +28,21 @@ The syntax for an arrow function expression does not require the `function` keyw
 
 Arrow functions can be defined with zero or more arguments. They can also be defined on one or more lines.
 
-> **Note:** Arrow functions functions are limited and can't be used in all situations:
->
+There are several variations of arrow functions :
```
Comment (yangc95) [inline content/javascript/concepts/arrow-functions/arrow-functions.md:31]:
```suggestion
There are several variations of arrow functions:
```

File: content/javascript/concepts/arrow-functions/arrow-functions.md [hunk 2]
```diff
@@ -33,16 +28,21 @@ The syntax for an arrow function expression does not require the `function` keyw
...
+- Arrow functions with a single parameter do not require `()` around the parameter list.
+- Arrow functions with a single expression (no curly brackets `{ }`) can use the concise function body which returns the result of the expression without the `return` keyword.
+
+## Limitations
+
+Arrow functions functions are limited and can't be used in all situations :
```
Comment (yangc95) [inline content/javascript/concepts/arrow-functions/arrow-functions.md:37]:
```suggestion
Arrow functions functions are limited and can't be used in all situations:

```

File: content/javascript/concepts/arrow-functions/arrow-functions.md [hunk 3]
```diff
@@ -33,16 +28,21 @@ The syntax for an arrow function expression does not require the `function` keyw
...
 > - They do not have their own bindings to [`this`](https://www.codecademy.com/resources/docs/javascript/this) or `super`, and should not be used as [methods](https://www.codecademy.com/resources/docs/javascript/methods).
 > - They cannot be used as [constructors](https://www.codecademy.com/resources/docs/javascript/constructors).
 > - They cannot use `yield`, within its body.
 > - They cannot use the special `arguments` keyword.
 
 ## Example
 
-The following example showcases how arrow functions are defined and applied:
+The following example showcases how arrow functions are defined and applied :
```
Comment (yangc95) [inline content/javascript/concepts/arrow-functions/arrow-functions.md:45]:
```suggestion
The following example showcases how arrow functions are defined and applied:
```

File: content/javascript/concepts/arrow-functions/arrow-functions.md [hunk 4]
```diff
@@ -57,7 +57,7 @@ printHello();
 checkWeight(25);
 ```
 
-This produces the following output:
+This produces the following output :
```
Comment (yangc95) [inline content/javascript/concepts/arrow-functions/arrow-functions.md:60]:
```suggestion
This produces the following output:
```

File: content/javascript/concepts/arrow-functions/arrow-functions.md [hunk 5]
```diff
@@ -66,10 +66,14 @@ Baggage weight: 25 kilograms.
 
 ## Codebyte Exammple
 
-The following example features an arrow function defined with two arguments on a single line:
+The following example features an arrow function defined with two arguments on a single line :
```
Comment (yangc95) [inline content/javascript/concepts/arrow-functions/arrow-functions.md:69]:
```suggestion
The following example features an arrow function defined with two arguments on a single line:
```

File: content/javascript/concepts/arrow-functions/arrow-functions.md [hunk 6]
```diff
@@ -66,10 +66,14 @@ Baggage weight: 25 kilograms.
...
-```js
+```codebyte/javascript
 const multiply = (a, b) => a * b;
 
 console.log(multiply(2, 30));
 ```
+This produces the following output :
+```shell
+60
+```
```
Comment (yangc95) [inline content/javascript/concepts/arrow-functions/arrow-functions.md:79]:
```suggestion
```

Comment (yangc95) [reply content/javascript/concepts/arrow-functions/arrow-functions.md:79]: Since this is a codebyte, we won't need the output explicitly stated

**Review by yangc95 — COMMENTED** [review]
@faiz887 Change requests were spot on for the most part! I just did a quick review for this. Don't hesitate to reach out and please tag me once you've looked over everything and resolved all comments. Thanks!

**Comment by faiz887** [comment]
@yangc95 I have resolved all the comments recommended. Please let me know if there's any other changes needed

**Review by SSwiniarski — APPROVED** [review]
@faiz887 looks good to me.

**Comment by Dusch4593** [comment]
@faiz887 I'm gonna take a look at this PR for review 3️⃣ 👀

File: content/javascript/concepts/arrow-functions/arrow-functions.md [hunk 7]
```diff
@@ -33,8 +28,14 @@ The syntax for an arrow function expression does not require the `function` keyw
 
 Arrow functions can be defined with zero or more arguments. They can also be defined on one or more lines.
```
Comment (Dusch4593) [inline content/javascript/concepts/arrow-functions/arrow-functions.md:29]:
```suggestion
Arrow functions can be defined with zero or more parameters, on one or more lines.
```

File: content/javascript/concepts/arrow-functions/arrow-functions.md [hunk 8]
```diff
@@ -33,8 +28,14 @@ The syntax for an arrow function expression does not require the `function` keyw
...
 
-> **Note:** Arrow functions functions are limited and can't be used in all situations:
->
+There are several variations of arrow functions:
+- Arrow functions with a single parameter do not require `()` around the parameter list.
+- Arrow functions with a single expression (no curly brackets `{ }`) can use the concise function body which returns the result of the expression without the `return` keyword.
+
```
Comment (Dusch4593) [inline content/javascript/concepts/arrow-functions/arrow-functions.md:34]:
```suggestion
In most cases, parentheses (`()`) are not required to be around the parameter list if there is one parameter. They are required in the following cases:
  - There is more than one parameter.
  - There are no parameters.
  - The single parameter is a destructured object.
  - There are default or rest parameters involved.
 
Arrow functions with a single expression have no curly brackets (`{ }`) and can use the concise function body to return the result of the expression without the `return` keyword. For multiple expressions, parentheses can be alternatively wrapped around the arrow function body to implicitly return the result.

```

File: content/javascript/concepts/arrow-functions/arrow-functions.md [hunk 9]
```diff
@@ -33,8 +28,14 @@ The syntax for an arrow function expression does not require the `function` keyw
...
+## Limitations
+
+Arrow functions functions are limited and can't be used in all situations:
+
 > - They do not have their own bindings to [`this`](https://www.codecademy.com/resources/docs/javascript/this) or `super`, and should not be used as [methods](https://www.codecademy.com/resources/docs/javascript/methods).
 > - They cannot be used as [constructors](https://www.codecademy.com/resources/docs/javascript/constructors).
 > - They cannot use `yield`, within its body.
```
Comment (Dusch4593) [inline content/javascript/concepts/arrow-functions/arrow-functions.md:41]:
```suggestion
Arrow functions can't be used in all situations dud to the following limitations:

- They do not have their own bindings to [`this`](https://www.codecademy.com/resources/docs/javascript/this) or `super`, and should not be used as [methods](https://www.codecademy.com/resources/docs/javascript/methods).
- They cannot be used as [constructors](https://www.codecademy.com/resources/docs/javascript/constructors).
- They cannot use `yield` within its body and create generators.
```

Comment (Dusch4593) [reply content/javascript/concepts/arrow-functions/arrow-functions.md:41]: Can we also remove the block quote and space from the line below? 😄

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@faiz887 left you some comments/suggestions to check out 😄

File: content/javascript/concepts/arrow-functions/arrow-functions.md [hunk 10]
```diff
@@ -33,8 +28,14 @@ The syntax for an arrow function expression does not require the `function` keyw
```
[the remark this answers is not in the document]
Comment (faiz887) [reply content/javascript/concepts/arrow-functions/arrow-functions.md:28]: Do you mean add it as a new point under the section "Limitations"?

File: content/javascript/concepts/arrow-functions/arrow-functions.md (hunk 9 above)
Comment (faiz887) [reply content/javascript/concepts/arrow-functions/arrow-functions.md]: Can you specify what space you want me to remove?

File: content/javascript/concepts/arrow-functions/arrow-functions.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (Dusch4593) [reply content/javascript/concepts/arrow-functions/arrow-functions.md:28]: Right 😄

**Review by Dusch4593 — APPROVED** [review]
@faiz887 This looks good to merge! 😄

I made a few edits in e684fd00b8b26ea73c18e3c3f7f1686d5f72f23d, including:

- Removed leading spaces in lists.
- On line # 27, I replaced the letters with numbers since the convention is 1..N, as opposed to A..N.

**Comment by Dusch4593** [comment]
Thank you for contributing to Codecademy Docs, @faiz887! 🎊

The changes should now be live: https://www.codecademy.com/resources/docs/javascript/arrow-functions

If you can't see your Codecademy profile under the list of contributors on the page, you may have to do one or both of the following:

- Add your GitHub username to your Codecdemy profile.
- Log out of Codecademy, and log back in through your GitHub account.

Looking forward to your next Docs contribution! 😄

**Pull request merged**